scalding-core/src/main/scala/com/twitter/scalding/Operations.scala

```diff
@@ -491,13 +497,23 @@ package com.twitter.scalding {
         .asScala
         .map(_.getObject(0).asInstanceOf[V])
 
+      val caches = values.toList
+      val numValuesPerKey = caches.size.toLong
+
       // Avoiding a lambda here
-      val resIter = reduceFnSer.get(key, values)
+      val resIter = reduceFnSer.get(key, caches.toIterator)
       while (resIter.hasNext) {
         val tup = Tuple.size(1)
-        tup.set(0, resIter.next)
+        val t2 = resIter.next
+
+        tup.set(0, t2)
         oc.add(tup)
       }
+      
+      flowProcess.increment(SkewMonitorCounters.KeyCount, SkewMonitorCounters.KeyCount, 1L)
```

Comment:
probably better to use a group like "scalding.debug" since I think we want to group them up so they are easy to see next to each other in the job tracker UI.